applications/StatisticsApplication/custom_methods/spatial_methods.h

```diff
@@ -356,45 +357,46 @@ class ContainerSpatialMethods
 
                 variance += std::pow(norm_method(current_value), 2);
             }
-#pragma omp atomic update
+#pragma omp atomic
             global_variance += variance;
         }
         global_variance =
             rModelPart.GetCommunicator().GetDataCommunicator().SumAll(global_variance);
-        const unsigned int number_of_items =
+        const double number_of_items =
             rModelPart.GetCommunicator().GetDataCommunicator().SumAll(
-                r_container.size());
+                static_cast<double>(r_container.size()));
```

Comment:
cannot you use `unsigned int` here too?

Reply:
i casted it directly to double since the next line I anyways have to deal with the double value, so i can avoid double casting.
`        return sum * (1.0 / std::max(number_of_items, 1.0));`.